types/src/cryptography/sig_hash.rs

```diff
@@ -0,0 +1,12 @@
+use crate::cryptography::Hash;
+
+pub const SIG_PREFIX_GAS_PRICE_ATTESTATION: &[u8] = b"ATT_GAS_PRICE";
+pub const SIG_VERSION_GAS_PRICE_ATTESTATION: u8 = 1;
+pub const SIG_PREFIX_TX_ATTESTATION: &[u8] = b"ATT_TX";
+pub const SIG_VERSION_TX_ATTESTATION: u8 = 1;
+pub const SIG_PREFIX_RECEIPT_ATTESTATION: &[u8] = b"ATT_RECEIPT";
+pub const SIG_VERSION_RECEIPT_ATTESTATION: u8 = 1;
+
+pub trait SigHashable {
+    fn hash_for_signature(&self) -> Hash;
```

Comment:
Should it also take a network-specifc prefix like this
```suggestion
    fn hash_for_signature(&self, network_prefix: &[u8]) -> Hash;
```

where this prefix would be different on each network (e.g. devnet, testnet, mainnet etc.) so that signatures signed for devnet cannot be re-used on other networks?